src/main/java/de/komoot/photon/nominatim/NominatimConnector.java

```diff
@@ -321,8 +321,12 @@ public void readEntireDatabase() {
 		final BlockingQueue<PhotonDoc> documents = new LinkedBlockingDeque<PhotonDoc>(20);
 		Thread importThread = new Thread(new ImportThread(documents));
 		importThread.start();
+		String countryCodeStr = "";
+		if(countryCodes.length > 0)
+			countryCodeStr = " AND country_code in ('" + String.join("','", countryCodes) +"')";
```

Comment:
I would lower-case the strings as Nominatim expects them this way. I would also suggest to do a bit of sanity checking of the parts: length must be exactly two characters and characters must be ascii letters only.